airconsole-1.3.0.js

```diff
@@ -132,21 +138,108 @@ AirConsole.prototype.onMessage = function(from, data) {};
 AirConsole.prototype.onCustomDeviceStateChange = function(device_id,
                                                           custom_data) {};
 
+
 /**
  * Gets called when a device joins/leaves a game session or updates its
  * DeviceState (custom DeviceState, profile pic, nickname, slow connection).
  * This is function is also called every time onConnect, onDisconnect or
  * onCustomDeviceStateChange is called. It's like their root function.
  * @abstract
- * @param {number} device_id - the device ID that changed it's DeviceState.
+ * @param {number} device_id - the device_id that changed its DeviceState.
  * @param user_data {AirConsole~DeviceState} - the data of that device.
  *        If undefined, the device has left.
  */
 AirConsole.prototype.onDeviceStateChange = function(device_id, device_data) {};
 
+
+/**
+ * Gets called when the screen sets the active players by calling
+ * setActivePlayers().
+ * @param {number|undefined} player_number - The player number of this device.
+ *                                           Can be undefined if this device
+ *                                           is not part of the active players.
+ */
+AirConsole.prototype.onActivePlayersChange = function(player_number) {};
+
+
+/**
+ * Takes all currently connected controllers and assigns them a player number.
+ * Can only be called by the screen. You don't have to use this helper
+ * function, but this mechanism is very convenient if you want to know which
+ * device is the first player, the second players, the third player ...
+ * The assigned player numbers always start with 0 and are consecutive.
+ * You can hardcode player numbers, but not device_ids.
+ * Once the screen has called setActivePlayers you can get the device_id of
+ * the first player by calling convertPlayerNumberToDeviceId(0), the device_id
+ * of the second player by calling convertPlayerNumberToDeviceId(1), ...
+ * You can also convert device_ids to player numbers by calling
+ * convertDeviceIdToPlayerNumber(device_id). You can get all device_ids that
+ * are active players by calling getActivePlayerDeviceIds().
+ * The screen can call this function every time a game round starts.
+ * @param {number} max_players - The maximum number of controllers that should
```

Comment:
Can we add for max_players: (optional) ... If undefined all connected controllers.